descartes_planner/src/planning_graph.cpp

```diff
@@ -630,45 +640,48 @@ bool PlanningGraph::getShortestPath(double &cost, std::list<JointTrajectoryPt>&
     vertex_index_map[i++] = dg_[jv].id;
   }
 
+  // Default the case to a very large number incase there is no valid path
+  // through the graph
   cost = std::numeric_limits<double>::max();
 
-  for (auto start = start_points.begin(); start != start_points.end(); ++start)
-  {
-    // initialize vectors to be used by dijkstra
-    std::vector<JointGraph::vertex_descriptor> predecessors(num_vert);
-    std::vector<double> weights(num_vert, std::numeric_limits<double>::max());
+  // initialize vectors to be used by dijkstra
+  std::vector<JointGraph::vertex_descriptor> predecessors(num_vert);
+  std::vector<double> weights(num_vert, std::numeric_limits<double>::max());
 
-    dijkstra_shortest_paths(
-        dg_,
-        *start,
-        weight_map(get(&JointEdge::transition_cost, dg_)).distance_map(
-          boost::make_iterator_property_map(weights.begin(), get(boost::vertex_index, dg_))).predecessor_map(
-          &predecessors[0]));
+  dijkstra_shortest_paths(
+      dg_,
+      virtual_vertex, // start from our fake point
+      weight_map(get(&JointEdge::transition_cost, dg_)).distance_map(
+        boost::make_iterator_property_map(weights.begin(), get(boost::vertex_index, dg_))).predecessor_map(
+        &predecessors[0]));
 
-    for (auto end = end_points.begin(); end != end_points.end(); ++end)
+  // Search the ending points for a minimum point
+  for (auto end = end_points.begin(); end != end_points.end(); ++end)
+  {
+    double weight = weights[*end];
```

Comment:
This is a bit confusing to me as the **weight** array is initialized to a size equal to **num_vert**, which I think is the number of trajectory points.  Shouldn't this array's size be set to the number of trajectories or end points?  Functionally, this probably doesn't impact the final result.